services/src/main/java/org/jboss/windup/web/services/rest/WindupEndpointImpl.java

```
@@ -58,31 +58,61 @@
      * @see org.jboss.windup.web.services.messaging.ExecutorMDB
      */
     @Override
-    public WindupExecution executeWithContext(Long contextId)
+    public WindupExecution executeProjectWithContext(AnalysisContext analysisContext, Long projectId)
     {
-        AnalysisContext analysisContext = this.analysisContextService.get(contextId);
-
-        if (analysisContext.getApplications().size() == 0)
+        if (analysisContext == null)
```

I think this should throw an exception if no analysis context is specified.